src/pinns_pde_solve.jl

```diff
@@ -282,80 +293,49 @@ function generate_training_sets(domains,dx,bcs,indvars,depvars,dict_indvars,dict
     spans = [d.domain.lower:dx:d.domain.upper for (d,dx) in zip(domains,dxs)]
     dict_var_span = Dict([Symbol(d.variables) => d.domain.lower:dx:d.domain.upper for (d,dx) in zip(domains,dxs)])
 
-    train_set = []
-    for points in Iterators.product(spans...)
-        push!(train_set, [points...])
-    end
+    train_set = map(points -> [points...], Iterators.product(spans...))
```

Comment:
why splat?

Reply:
replaced with `collect(points)` instead of splat.

Reply:
what is `points` here: a tuple?

Reply:
yes

Reply:
but actually, splat is faster

Reply:
Really? Show how: the two should be the same if it's a small tuple. But the bigger thing is, why are tuples generated? It should probably go straight to arrays.

Reply:
yes, we should go straight to arrays it would be the best but `Iterators.product(spans...)` create a tuple.

Reply:
```julia
    spans = [0.0:0.25:2.0, 0.0:0.25:2.0, 0.0:0.25:2.0]
    julia> @benchmark map(points -> collect(points), Iterators.product(spans...))
    BenchmarkTools.Trial:
      memory estimate:  86.42 KiB
      allocs estimate:  737
      --------------
      minimum time:     51.168 μs (0.00% GC)
      median time:      59.035 μs (0.00% GC)
      mean time:        88.805 μs (21.89% GC)
      maximum time:     68.412 ms (99.81% GC)
      --------------
      samples:          10000
      evals/sample:     1

    julia> @benchmark map(points -> [points...], Iterators.product(spans...))
    BenchmarkTools.Trial:
      memory estimate:  86.42 KiB
      allocs estimate:  737
      --------------
      minimum time:     48.730 μs (0.00% GC)
      median time:      66.870 μs (0.00% GC)
      mean time:        158.667 μs (13.34% GC)
      maximum time:     76.640 ms (99.82% GC)
      --------------
      samples:          10000
      evals/sample:     1
    ```
    uh,  `collect` is a little faster.